homeassistant/components/ryse/cover.py

```diff
@@ -0,0 +1,129 @@
+        """Update cover position when receiving notification."""
+        if 0 <= position <= 100:
+            self._current_position = 100 - position
+            self._state = "open" if position < 100 else "closed"
+            _LOGGER.debug("Updated cover position: %02X", position)
+        self.async_write_ha_state()
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the shade."""
+        pdata = build_position_packet(0x00)
+        await self._device.write_data(pdata)
+        _LOGGER.debug("Binary packet to change position to open")
+        self._state = "open"
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the shade."""
+        pdata = build_position_packet(0x64)
+        await self._device.write_data(pdata)
+        _LOGGER.debug("Binary packet to change position to close")
+        self._state = "closed"
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Set the shade to a specific position."""
+        position = 100 - kwargs.get("position", 0)
+        pdata = build_position_packet(position)
+        await self._device.write_data(pdata)
+        _LOGGER.debug("Binary packet to change position to a specific position")
+        self._state = "open" if position < 100 else "closed"
+
+    async def async_update(self) -> None:
+        """Fetch the current state and position from the device."""
+        if not self._device.client or not self._device.client.is_connected:
+            paired = await self._device.pair()
+            if not paired:
+                _LOGGER.warning("Failed to pair with device, skipping update")
+                return
+
+        try:
+            if self._current_position is None:
+                bytesinfo = build_get_position_packet()
+                await self._device.write_data(bytesinfo)
+        except (TimeoutError, OSError) as err:
+            _LOGGER.error("BLE communication error while reading device data: %s", err)
+        except Exception:
+            _LOGGER.exception("Unexpected error while reading device data")
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return True if the shade is closed."""
+        return self._state == "closed"
+
+    @property
+    def current_cover_position(self) -> int | None:
+        """Return current cover position."""
+        if self._current_position is None:
+            return 50
+        if not (0 <= self._current_position <= 100):
+            _LOGGER.warning(
+                "Invalid position value detected: %02X",
+                self._current_position,
+            )
+            return 50
+        return int(self._current_position)
+
+    @property
+    def supported_features(self) -> CoverEntityFeature:
+        """Return supported features for the cover entity."""
+        return (
+            CoverEntityFeature.OPEN
+            | CoverEntityFeature.CLOSE
+            | CoverEntityFeature.SET_POSITION
+        )
```

Comment:
Instead of a property add this at the top of the class
```suggestion
    _attr_supported_features = (
            CoverEntityFeature.OPEN
            | CoverEntityFeature.CLOSE
            | CoverEntityFeature.SET_POSITION
        )
```